content/en/blog/2020/on-demand-egress-gateways/index.md

```diff
@@ -0,0 +1,500 @@
+{{< /tip >}}
+
+Create a `namespace` where you will install the `Egress Gateway`
+
+{{< tip >}}
+Remember that we assume *namespace = httpbin*)
+{{< /tip >}}
+
+{{< text bash >}}
+$ kubectl create ns httpbin
+{{< /text >}}
+
+As it is described in the [documentation](/docs/setup/install/istioctl/#customize-kubernetes-settings), you can deploy several `IstioOperator` resources. However, they have to be pre-parsed and then applied to the cluster.
+
+{{< text bash >}}
+$ istioctl manifest generate -f <path-to-egress-file> | kubectl apply -f -
+{{< /text >}}
+
+Where `<path-to-egress-file>` is the path where you have saved the new `IstioOperator` CR.
+
+### Istio configuration
+
+Now you will set up `Istio` to connect to the external service [https://httpbin.org](https://httpbin.org)
+
+#### Certificate for TLS
+
+How to generate the certificate is explained in the [documentation](/docs/tasks/traffic-management/ingress/secure-ingress/#generate-client-and-server-certificates-and-keys).
+
+{{< text bash >}}
+$ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=example Inc./CN=<my-hostname>' -keyout example.com.key -out example.com.crt
+{{< /text >}}
+
+{{< text bash >}}
+$ openssl req -out httpbin.example.com.csr -newkey rsa:2048 -nodes -keyout httpbin.example.com.key -subj "/CN=<my-hostname>/O=httpbin organization"
+{{< /text >}}
+
+{{< text bash >}}
+$ openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in httpbin.example.com.csr -out httpbin.example.com.crt
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl create -n istio-system secret tls <my-secret-name> --key=httpbin.example.com.key --cert=httpbin.example.com.crt
+{{< /text >}}
+
+Where `<my-secret-name>` is the name used later for the `Gateway` resource. `<my-hostname>` is the hostname used to access your service.
+
+#### Ingress Gateway
+
+Create a `Gateway` resource to operate ingress gateway to accept requests.
+
+{{< warning >}}
+Make sure that only one `Gateway` spec matches the hostname. `Istio` gets confused when there are multiple Gateway definitions covering the same hostname.
+{{< /warning >}}
+
+An example:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: my-ingressgateway
+  namespace: istio-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - hosts:
+    - "<my-hostname>"
+    port:
+      name: http
+      number: 80
+      protocol: HTTP
+    tls:
+     httpsRedirect: true
+  - port:
+      number: 443
+      name: https
+      protocol: https
+    hosts:
+    - "<my-hostname>"
+    tls:
+      mode: SIMPLE
+      credentialName: <my-secret-name>
+{{< /text >}}
+
+Where `<my-hostname>` is the hostname to access the service through the `my-ingressgateway` and `<my-secret-name>` is the secret which contains the certificate.
+
+#### Egress Gateway
+
+Create another Gateway object, but this time to operate the `Egress Gateway` you have already installed with `helm`:
```

Comment:
I haven't installed anything with Helm yet?

Reply:
From the version for istio 1.7. Sorry. Deleted